### ISSUE TITLE
'cellValueChanged' event triggers every time when cell editing stops.

### ISSUE THREAD

**Issue by EoinGriffin-AI** [ag-grid/ag-grid#2592]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

I've figured out the cause of this bug, and found the commit where it was introduced. See `Cause` below.

**Providing a Reproducible Scenario**
Double-click on a cell, do NOT make any change, then press `enter`.
The `onCellValueChanged` event triggers every time a cell stops editing, even when you don't make any change. In the plunker below, you'll see an alert that's sent from that event handler.
- https://plnkr.co/edit/LaLSbQBIwWEwXGM6oH7k?p=preview

**Current behavior**
onCellValueChanged event triggers 100% of the time, with `newValue` and `oldValue` being the same. Event triggers regardless of whether or not there's any change.

**Expected behavior**
Event should only trigger if there's a change.

**Cause**
I've tracked down the problem to this code:
https://github.com/ag-grid/ag-grid/blob/master/packages/ag-grid/src/ts/valueService/valueService.ts#L167-L186
^^ As you can see, this function will always return `true`, and it should definitely not.
The functionality existed, but was removed by "niall" (@ceolter ?) in June 2017 (why!?):
https://github.com/ag-grid/ag-grid/commit/9ecd4c34d5299dfa171528e8f89dea16b6c7cb80#diff-9ec17707a22c3c4886ac3a7b2b3b88cc
The code that depends on this value being sensible still exists in the codebase here, and without correctly detecting a change to the cell then the 'cellValueChanged' event is totally redundant (might as well just use the 'cellEditingStopped' event... but... please fix this bug instead? Restore that old code?:
https://github.com/ag-grid/ag-grid/blob/master/packages/ag-grid/src/ts/valueService/valueService.ts#L120

**Please tell us about your environment:**
Windows 7

* **ag-Grid version:**
18.1.2 (I've tested this with older versions of ag-Grid too, and it's been like this for some time. Since June 29th 2017, to be precise.

* **Browser:**
Chrome

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

**Comment by EoinGriffin-AI** [comment]
@makinggoodsoftware I respectfully disagree. The event name is misleading and does not work as it should. If you look at the code and the history of the code, it's clearly an unintended bug and now you're saying "Yeah, but the bug might be useful."
Sure it might be, but you're prioritizing the new bug behaviour higher than the intended behaviour.
If you now want that new behaviour then add another event with a name that correctly reflects what it does... but fix the bug.

This bug and your decision to not fix it requires users to know that they need to use a workaround to get the intended behaviour. Frankly, that's just lazy. I get it, I'm a software developer too, and laziness is pretty much an art form for us. If it's not laziness, then ignoring the issue is not the right solution.

If the new behavior is what's now expected, then the event should be renamed to something like 'cellEditingStopped', but as it stands right now it definitely does not do what it should be doing.
And at the very least update the documentation so that others won't waste their time trying to figure out what's going on with this.

**Comment by SmokeyPHP** [comment]
Totally agree, functionality does not match the expected outcome - even the documentation describes the event as "Value has changed after editing"

**Comment by igpeev** [comment]
In addition to the upper arguments, I would add that in (cellValueChanged) event object ag-grid changes the value in 'newValue' property from number to string at it's own discretion, so  === will always miss if a number is updated. For example value was 5 (in ev.oldValue) , now it is "5" (in ev.newValue), hence 5==="5" always yields false. Why this conversion, can you just leave a number to stay number ?

**Comment by fdtaljaard** [comment]
I agree the event should fire on Value changed as per the name.

**Comment by ivankokan** [comment]
> In addition to the upper arguments, I would point out that in (cellValueChanged) event object ag-grid changes the value in 'newValue' property from number to string at it's own discretion, so === will always miss if a number is updated. For example if old value was 5 (in ev.oldValue) , and now the new value is "5" (in ev.newValue), hence 5==="5" always yields false. Why this conversion, can you just leave a number stay a number ?

Is there a simple way to overcome what @igpeev mentioned?
